server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -686,6 +687,105 @@ public void testConcurrentSnapshotCreateAndDeleteOther() {
+            .snapshots(snapshotsList)
+            .ignoreUnavailable(true);
+
+        TransportGetSnapshotsAction transportGetSnapshotsAction = (TransportGetSnapshotsAction) getSnapshotsAction;
+        transportGetSnapshotsAction.execute(null, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+            assertNotNull("Snapshot list should be registered", repoSnapshotResponse.getSnapshots());
+            assertThat(repoSnapshotResponse.getSnapshots(), hasSize(2));
+        }, exception -> { throw new AssertionError(exception); }));
+    }
+
+    public void testTransportGetCurrentSnapshotOnly() {
+        setupTestCluster(randomFrom(1, 3, 5), randomIntBetween(2, 10));
+
+        String repoName = "repo";
+        String snapshotName = GetSnapshotsRequest.CURRENT_SNAPSHOT;
+        final String index = "test";
+        final int shards = randomIntBetween(1, 10);
+
+        TestClusterNodes.TestClusterNode clusterManagerNode = testClusterNodes.currentClusterManager(
+            testClusterNodes.nodes.values().iterator().next().clusterService.state()
+        );
+
+        final StepListener<CreateSnapshotResponse> createSnapshotResponseStepListener = new StepListener<>();
+
+        continueOrDie(
+            createRepoAndIndex(repoName, index, shards),
+            createSnapshotResponse -> client().admin()
+                .cluster()
+                .prepareCreateSnapshot(repoName, snapshotName)
+                .setWaitForCompletion(true)
+                .execute(createSnapshotResponseStepListener)
+        );
+        final StepListener<GetSnapshotsResponse> getSnapshotsResponseStepListener = new StepListener<>();
+        continueOrDie(getSnapshotsResponseStepListener, getSnapshotsResponse -> {
+            client().admin().cluster().prepareGetSnapshots(repoName).execute(getSnapshotsResponseStepListener);
+        });
+        deterministicTaskQueue.runAllRunnableTasks();
+
+        final Repository repository = clusterManagerNode.repositoriesService.repository(repoName);
+        Collection<SnapshotId> snapshotIds = getRepositoryData(repository).getSnapshotIds();
+        assertThat(snapshotIds, hasSize(0));
+
+        final String[] snapshotsList = { GetSnapshotsRequest.CURRENT_SNAPSHOT };
+        TransportAction getSnapshotsAction = clusterManagerNode.actions.get(GetSnapshotsAction.INSTANCE);
+        GetSnapshotsRequest repoSnapshotRequest = new GetSnapshotsRequest().repository(repoName)
+            .snapshots(snapshotsList)
+            .ignoreUnavailable(true);
+
+        TransportGetSnapshotsAction transportGetSnapshotsAction = (TransportGetSnapshotsAction) getSnapshotsAction;
+        transportGetSnapshotsAction.execute(null, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+            assertNotNull("Empty Snapshot info should be registered for current snapshots", repoSnapshotResponse.getSnapshots());
+            assertThat(repoSnapshotResponse.getSnapshots(), hasSize(0));
```

Comment:
how is the current snapshot tested here. I see you are asserting for size 0?

Reply:
@shwetathareja This is to test the below codecoverage checks:
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L147
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L156

However, this would only be populated to snapshotInfo if the `_CURRENT` snapshot is in InProgress state: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java#L156, This has not been mocked here. This test can be removed altogether if not required.

Reply:
we shouldn't add tests just for code coverage sake if they are not testing intended functionality.

Reply:
Ack.